dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -78,20 +78,20 @@ const (
 )
 
 func init() {
-	registry.RegisterModel(krm.{{.Kind}}GVK, NewModel)
+	registry.RegisterModel(krm.{{.Kind}}GVK, New{{.Kind}}Model)
 }
 
-func NewModel(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
-	return &model{config: *config}, nil
+func New{{.Kind}}Model(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
+	return &model{{.Kind}}{config: *config}, nil
 }
 
-var _ directbase.Model = &model{}
+var _ directbase.Model = &model{{.Kind}}{}
 
-type model struct {
+type model{{.Kind}} struct {
 	config config.ControllerConfig
 }
 
-func (m *model) client(ctx context.Context) (*gcp.Client, error) {
```

Comment:
nit: `clientForKind` seems a bit verbose? Given that the function already has a pointer receiver of `modelKind`.